### PR TITLE
#159 Order Books endpoint

### LINKED ISSUE #159: Order Books endpoint
ℹ️ This is part of the Epic #158 

## Context
An order book is a collections of offers for a specific pair of assets.
https://developers.stellar.org/api/aggregations/order-books/object/

## Objective
* To get access to all the Order Books endpoints exposed by Horizon.

## Checklist
- [x] Retrieve an order book

### PR DESCRIPTION
**Pull request by MrJuanGaviriaK**
Closes #159 

## Objective
* To get access to all the Order Books endpoints exposed by Horizon.

## Checklist
- [x] Retrieve an order book

## Test coverage
Screen Shot 2022-07-21 at 3 27 47 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -703,6 +704,24 @@ Stellar.Horizon.PaymentPaths.list_send_paths(source_asset_type: :native, source_
...
+- [Required] buying_asset_type. The type for the asset being bought (counter asset), **native**, **credit_alphanum4**, or **credit_alphanum12**.
+- [Optional] selling_asset_issuer. The Stellar address of the issuer of the asset being sold (base asset). Required if the selling_asset_type is not native
+- [Optional] selling_asset_code. The code for the asset being sold (base asset). Required if the selling_asset_type is not native.
+- [Optional] buying_asset_issuer. The Stellar address of the issuer of the asset being bought (counter asset). Required if the **buying_asset_type** is not **native**.
+- [Optional] buying_asset_code. The code for the asset being bought (counter asset). Required if the **buying_asset_type** is not **native**.
+- [Optional] limit. The maximum number of records returned
+
+```elixir
+Stellar.Horizon.OrderBooks.retrieve(selling_asset_type: :native, buying_asset_type: :native)
+```
```
Comment (juanhurtado10) [inline README.md:721]: @MrJuanGaviriaK to make the endpoints easy to use, we definitely can improve the definition for this endpoint interface.

**My proposal:**

```elixir
Stellar.Horizon.OrderBooks.retrieve(selling_asset: :native, buying_asset: :native, limit: 20)
Stellar.Horizon.OrderBooks.retrieve(selling_asset: :native, buying_asset: :native, limit: 10)
...
Stellar.Horizon.OrderBooks.retrieve(selling_asset: [code: "BTC", issuer: "ISSUER"], buying_asset: :native, limit: 10)
```

Internally, you can resolve the required params for the endpoints, and in this way, provide a better UX.

What do you think?

File: lib/horizon/order_book/price.ex [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+  defstruct [
+    :price_r,
+    :price,
+    :amount
+  ]
+
+  @price_r_mapping [
+    n: :integer,
```
Comment (juanhurtado10) [inline lib/horizon/order_book/price.ex:23]: @MrJuanGaviriaK what is a `price_r_mapping`? IMO, you can remove this constant and place it directly in `@mapping`

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]
@MrJuanGaviriaK You've done an outstanding job 🚀. Thanks for taking the time and contribute to this. 

Minor changes were requested 😃.

File: README.md (hunk 1 above)
Comment (MrJuanGaviriaK) [reply README.md:721]: I agree, I'm going to improve it, thanks!

File: lib/horizon/order_books.ex [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  ## Parameters
+    * `selling_asset_type`: The type for the asset being sold (base asset), native, credit_alphanum4, or credit_alphanum12
+    * `buying_asset_type`: The type for the asset being bought (counter asset), native, credit_alphanum4, or credit_alphanum12
+
+  ## Options
+    * `selling_asset_issuer`: The Stellar address of the issuer of the asset being sold (base asset). Required if the selling_asset_type is not native
+    * `selling_asset_code`: The code for the asset being sold (base asset). Required if the selling_asset_type is not native.
+    * `buying_asset_issuer`: The Stellar address of the issuer of the asset being bought (counter asset). Required if the buying_asset_type is not native.
+    * `buying_asset_code`: The code for the asset being bought (counter asset). Required if the buying_asset_type is not native.
+    * `limit`: The maximum number of records returned
```
Comment (juanhurtado10) [inline lib/horizon/order_books.ex:31]: @MrJuanGaviriaK after changes these docs may not apply

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

**Review by juanhurtado10 — APPROVED** [review]
@MrJuanGaviriaK Perfection 🛸

**Pull request merged**